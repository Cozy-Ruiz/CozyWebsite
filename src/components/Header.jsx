import React, { useState, useContext } from 'react';
import Link from 'next/link';
import Image from 'next/image';
import MenuUser from './MenuUser';
import MenuNav from './MenuNav';
import menu from '@icons/icon_menu.svg';
import logo from '@logos/LogoCWS.png';
//import AppContext from '@context/AppContext';
//import shoppingCart from '@icons/icon_shopping_cart.svg';
import user from '@icons/icon_user.png';
import styles from '@styles/Header.module.scss';

import { signIn, signOut, useSession } from 'next-auth/react';

const Header = () => {
	const { data: session, status } = useSession()

	const [toggleMenu, setToggleMenu] = useState(false);
	const [toggleUser, setToggleUser] = useState(false);
	//const { state } = useContext(AppContext);

	const handleToggleMenu = () => {
		setToggleMenu(!toggle);
	}

	const handleToggleUser = () => {
		setToggleUser(!toggle);
	}

	return (
		<>
		<nav className={styles.nav}>
			<Image src={menu} alt="menu" className={styles.menu} onClick={() => setToggleMenu(!toggleMenu)}/>
			<div className={styles['navbar-left']}>
				<Link href="/" >
					<Image src={logo} alt="logo" className={styles['nav-logo']} />
				</Link>
				
				<MenuNav/>
			</div>
			<div className={styles['navbar-right']}>
				<ul>
					
					<li className={styles['navbar-email']}>
						{session && session.user && session.user.name}
					</li>
					
					<li className={styles['navbar-shopping-cart']} onClick={() => setToggleUser(!toggleUser)}>
						<Image style={{width: '25px', height: '25px'}} src={user} alt="user" />
					</li>
				</ul>
			</div>
		</nav>
		{toggleMenu && <MenuNav />}
		{toggleUser && <MenuUser />}
		</>
	);
}

export default Header;