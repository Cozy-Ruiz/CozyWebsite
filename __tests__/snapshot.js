/**
 * @jest-environment jsdom
 */

import React from 'react';
import { SessionProvider } from "next-auth/react"; // Import SessionProvider
import { render } from "@testing-library/react";
import Header from '../src/components/Header';
 
it('renders Header unchanged', () => {
  const { container } = render(
    <SessionProvider session={null}>
      <Header />
    </SessionProvider>
  )
  expect(container).toMatchSnapshot()
})