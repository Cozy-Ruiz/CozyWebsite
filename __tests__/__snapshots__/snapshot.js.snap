// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header unchanged 1`] = `
<div>
  <nav>
    <img
      alt="menu"
      data-nimg="1"
      decoding="async"
      loading="lazy"
      src=""
      style="color: transparent;"
    />
    <div>
      <a
        href="/"
      >
        <img
          alt="logo"
          data-nimg="1"
          decoding="async"
          loading="lazy"
          src=""
          style="color: transparent;"
        />
      </a>
      <ul>
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/about"
          >
            About
          </a>
        </li>
        <li>
          <a
            href="/projects"
          >
            Projects
          </a>
        </li>
        <li>
          <a
            href="/contact"
          >
            Contact
          </a>
        </li>
      </ul>
    </div>
    <div>
      <ul>
        <li />
        <li>
          <img
            alt="user"
            data-nimg="1"
            decoding="async"
            loading="lazy"
            src=""
            style="color: transparent; width: 25px; height: 25px;"
          />
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
